uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+        cfg = {'entitlement': {'directives': directives}}
+        with mock.patch('uaclient.util.subp') as m_subp:
+            process_directives(cfg)
+        assert 0 == m_subp.call_count
+
+
+class TestLivepatchEntitlementCanEnable(TestCase):
+
+    @mock.patch('uaclient.util.is_container', return_value=False)
+    @mock.patch('os.getuid', return_value=0)
+    def test_can_enable_true_on_entitlement_inactive(
+            self, m_getuid, m_is_container):
+        """When operational status is INACTIVE, can_enable returns True."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            self.assertTrue(entitlement.can_enable())
+        self.assertEqual('', m_stdout.getvalue())
+
+    @mock.patch('uaclient.util.is_container', return_value=True)
+    @mock.patch('os.getuid', return_value=0)
+    def test_can_enable_false_on_containers(self, m_getuid, m_is_container):
+        """When in a is_container is True, can_enable returns False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            entitlement = LivepatchEntitlement(cfg)
+            self.assertFalse(entitlement.can_enable())
+        msg = 'Cannot install Livepatch on a container\n'
+        assert msg == m_stdout.getvalue()
+
+
+class TestLivepatchEntitlementEnable(TestCase):
+
+    with_logs = True
+
+    mocks_install = [mock.call(
+        ['snap', 'install', 'canonical-livepatch'], capture=True)]
+    mocks_config = [
+        mock.call(
+            ['/snap/bin/canonical-livepatch', 'config',
+             'remote-server=https://alt.livepatch.com'], capture=True),
+        mock.call(
+            ['/snap/bin/canonical-livepatch', 'enable', 'TOKEN'], capture=True)
+    ]
+
+    @mock.patch(M_PATH + 'LivepatchEntitlement.can_enable', return_value=False)
+    def test_enable_false_when_can_enable_false(self, m_can_enable):
+        """When can_enable returns False enable returns False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
```

Comment:
Should this test be asserting something about `m_stdout`? (If not, we don't need `as m_stdout` here.)

Reply:
Validated m_stdout isn't being saturated w/ print statements.